sdk/appplatform/mgmt/src/main/java/com/azure/resourcemanager/appplatform/implementation/SpringServiceImpl.java

```diff
@@ -183,4 +193,39 @@ public SpringServiceImpl withoutGitConfig() {
     protected Mono<ServiceResourceInner> getInnerAsync() {
         return manager().inner().getServices().getByResourceGroupAsync(resourceGroupName(), name());
     }
+
+    @Override
+    public SpringServiceImpl withCertificate(String name, String keyVaultUri, String certNameInKeyVault) {
+        this.addPostRunDependent(
+            context -> certificates.createOrUpdateAsync(
+                name,
+                new CertificateProperties().withVaultUri(keyVaultUri).withKeyVaultCertName(certNameInKeyVault)
+            ).cast(Indexable.class)
+        );
+        return this;
+    }
+
+    @Override
+    public SpringServiceImpl withCertificate(String name, String keyVaultUri,
+                                             String certNameInKeyVault, String certVersion) {
+        this.addPostRunDependent(
+            context -> certificates.createOrUpdateAsync(
+                name,
+                new CertificateProperties()
+                    .withVaultUri(keyVaultUri)
+                    .withKeyVaultCertName(certNameInKeyVault)
+                    .withCertVersion(certVersion)
+            ).cast(Indexable.class)
+        );
+        return this;
+    }
+
+    @Override
+    public SpringServiceImpl withoutCertificate(String name) {
+        this.addPostRunDependent(
+            context -> certificates.deleteByNameAsync(name)
+                .then(context.voidMono())
+        );
+        return this;
+    }
```

Comment:
Could be a problem if certificate with `name` does not exist. Though seems no better approach.

Reply:
It seems the deletion would be successful when the `name` does not exist.